document_ai_warehouse_processing_python/document_warehouse_utils.py

```diff
@@ -0,0 +1,377 @@
+        elif "property_type_options" in property_definition:
+            prop.property_type_options = property_definition["property_type_options"]
+        elif "date_time_type_options" in property_definition:
+            prop.date_time_type_options = property_definition["date_time_type_options"]
+        elif "enum_type_options" in property_definition:
+            prop.enum_type_options = property_definition["enum_type_options"]
+        elif "map_type_options" in property_definition:
+            prop.map_type_options = property_definition["map_type_options"]
+
+        return prop
+
+    def create_document_schema(self, schema) -> contentwarehouse_v1.DocumentSchema:
+
+        schema_json = json.loads(schema)
+
+        client = self.get_document_schema_service_client()
+        parent = client.common_location_path(self.project_number, self.api_location)
+
+        # Initialize request argument(s)
+        request = contentwarehouse_v1.CreateDocumentSchemaRequest()
+
+        request.parent = parent
+
+        # define schema
+        request.document_schema = contentwarehouse_v1.DocumentSchema()
+        request.document_schema.display_name = schema_json["display_name"]
+        request.document_schema.document_is_folder = schema_json["document_is_folder"]
+        request.document_schema.description = schema_json["description"]
+
+        for property_definition in schema_json["property_definitions"]:
+            request.document_schema.property_definitions.append(self.create_document_schema_property_definition(property_definition))
+
+        print(request)
+        print(request.document_schema)
+
+        #Make the request
+        response = client.create_document_schema(request=request)
+
+        # Handle the response
+        return response
+
+    def get_document_schema(self,schema_id: str):
+        client = self.get_document_schema_service_client()
+        parent = client.common_location_path(self.project_number, self.api_location)
+
+        # Initialize request argument(s)
+        request = contentwarehouse_v1.GetDocumentSchemaRequest()
+
+        request.name = f"{parent}/documentSchemas/{schema_id}"
+
+        # Make the request
+        response = client.get_document_schema(request=request)
+
+        # Handle the response
+        return response
+
+    def delete_document_schema(self,schema_id: str):
+
+        client = self.get_document_schema_service_client()
+        parent = client.common_location_path(self.project_number, self.api_location)
+
+        # Initialize request argument(s)
+        request = contentwarehouse_v1.DeleteDocumentSchemaRequest()
+
+        request.name = f"{parent}/documentSchemas/{schema_id}"
+
+        # Make the request
+        response = client.delete_document_schema(request=request)
+
+        # Handle the response
+        return response
+
+    def update_document_schema(self, schema_id: str, schema: str):
+        schema_json = json.loads(schema)
```

Comment:
Same here with json conversion.

There are parameters you can add to control if it ignores unknown fields if needed

`ignore_unknown_fields: bool = False`

Reply:
Changed to contentwarehouse_v1.DocumentSchema.from_json() for json conversion